src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1035,7 +1035,7 @@ async Task<int> ExecuteNonQuery(bool async, CancellationToken cancellationToken)
             using var reader = await ExecuteReaderAsync(CommandBehavior.Default, async, cancellationToken);
             while (async ? await reader.NextResultAsync(cancellationToken) : reader.NextResult()) ;
 
-            reader.Close();
+            await reader.CloseAsync();
```

Comment:
Since this method is used for both sync and async (see the async parameter above), this would make the sync path work over async, which is really not good.

Given that the while loop just above drains the command's results, Close here should be 100% synchronous and not perform any I/O, so it's more correct to leave things as they were (though a comment could indeed clarify that).

Reply:
done. should the above `using var reader = await ExecuteReaderAsync` be sync as well?

Reply:
Nope. Although that method has the Async suffix, it receives the async flag parameter which instructs it whether to actually be sync or not. That flag is passed throughout the stack, and when we get to the point of actually doing I/O, we branch and either do a sync or async operation. This allows Npgsql to have a single code path for both sync and async.